Feat/user meta edit UI

hakanonal opened the pull request.

hakanonal committed:
feat: edit user settings meta view UI

hakanonal committed:
feat: edit user ui error messages

hakanonal committed:
feat: update user meta form to db write error

EeroHeikkinen commented:
Looking like a decent start! Can access the settings screen in the dashboard and edit the settings. 

I guess the next steps would be related to popping up the new screen asking to correct / confirm this information after the first login of the user.

hakanonal commented:
@EeroHeikkinen anna do you have any kind of UI related code that we can quicklly apply to the form so that the form looks nicer. I am very bad at UI it generally tooks so much time of mine.

hakanonal committed:
feat: nicer look of the form

hakanonal committed:
fix: changed scope of bootstrap stylesheet

EeroHeikkinen commented on src/app.controller.ts:
```diff
@@ -597,4 +601,49 @@ export class AppController {
+      throw new BadRequestException("This E-Mail is being used by another user!");
+    
+    const userToUpdate = plainToClassFromExist(user,body);
+    userToUpdate.emails.push(body.email);
```
Could we also save this into a platform connection object? This would make the logic of data gathered from earth points consistent with data such as emails from other platforms.

The call to create the first platform connection:

```
    await this.platformConnectionService.create({
      userid,
      profile_id: userid,
      platform: 'earthpoints',
      auth_token: null,
      auth_expiration: null,
      emails: [body.email],
    });
```

If a platform connection object already exists for earthpoints we would need to retrieve it and update it instead.

hakanonal committed:
feat: implemented overlay instead of seperate page

- TODO: app controller must be arragned again.
- TODO: post end point must ajax based in order to manage error experience.

hakanonal committed:
feat: implemented save functionality to save

hakanonal committed:
Merge branch 'main' into feat/user-meta-edit-ui

hakanonal committed:
feat: lastName column to users

hakanonal committed:
feat: getting phone from India users.

hakanonal reviewed (commented):
@EeroHeikkinen anna can you review this last commit. especially the changes on app.controller. I have saved the phone number as platform. I could not contamplete how I am going to merge it.

I think phone number is not needed to be uniqe though out the users. but e-mail should be unique.

hakanonal committed:
feat: jquery validation plugin

hakanonal committed:
fix: ui experiences
